test/files/jvm/scala-concurrent-tck.scala

```diff
@@ -541,6 +538,26 @@ def testTransformFailure(): Unit = once {
     check(Future.failed[Int](new Exception))
   }
 
+  private[this] final def testMulti(f: Future[String] => Future[String]): Unit = {
+    val p = Promise[String]()
+    val f1, f2, f3 = f(p.future)
```

Comment:
@richdougherty Verifying that many dependencies/links on the same Promise works.